app/src/main/java/echo/SocketIO.java

```diff
@@ -0,0 +1,24 @@
+package echo;
+
+import java.io.*;
+import java.net.Socket;
+
+public class SocketIO {
+
+  public String readMessage(Socket serverConnection) {
+    String string = "";
```

Comment:
We usually don't call the variable name the same as the type. "String string" here for example. One standard I've seen, which helps when you have a challenge trying to think of a good variable name, is to call the value "result". It is a way to tell the user that this is the value that the function returns so "pay attention to how it changes".

Reply:
This makes so much sense. Thanks for pointing that out!